client/js/widgets/button.js

```diff
@@ -28,452 +24,6 @@ export class Button extends Widget {
-            return w.p(a.property) === a.value;
-          }).slice(0, a.max).concat(a.mode == 'add' ? collections[a.collection] || [] : []);
-
-          // resolve piles
-          c.filter(w=>w.p('type')=='pile').forEach(w=>c.push(...w.children()));
-          c = c.filter(w=>w.p('type')!='pile');
-          collections[a.collection] = c;
-        }
-      }
-
-      if(a.func == 'SET') {
-        setDefaults(a, { collection: 'DEFAULT', property: 'parent', relation: '=', value: null });
-        if((a.property == 'parent' || a.property == 'deck') && a.value !== null && !widgets.has(a.value)) {
-          problems.push(`Tried setting ${a.property} to ${a.value} which doesn't exist.`);
-        } else if(isValidCollection(a.collection)) {
-          if([ '+', '-', '=' ].indexOf(a.relation) == -1)
-            problems.push(`Warning: Relation ${a.relation} interpreted as =.`);
-          for(const w of collections[a.collection]) {
-            if(a.relation === '+')
-              w.p(a.property, w.p(a.property) + a.value);
-            else if(a.relation === '-')
-              w.p(a.property, w.p(a.property) - a.value);
-            else
-              w.p(a.property, a.value);
-          }
-        }
-      }
-
-      if(a.func == 'SORT') {
-        setDefaults(a, { key: 'value', reverse: false });
-        if(isValidID(a.holder)) {
-          this.w(a.holder, holder=>{
-            let z = 1;
-            let children = holder.children().reverse().sort((w1,w2)=>{
-              if(typeof w1.p(a.key) == 'number')
-                return w1.p(a.key) - w2.p(a.key);
-              else
-                return w1.p(a.key).localeCompare(w2.p(a.key));
-            });
-            if(a.reverse)
-              children = children.reverse();
-            children.forEach(c=>c.p('z', ++z));
-            holder.updateAfterShuffle();
-          });
-        }
-      }
-
-      if(a.func == 'SHUFFLE') {
-        if(isValidID(a.holder)) {
-          this.w(a.holder, holder=>{
-            holder.children().forEach(c=>c.p('z', Math.floor(Math.random()*10000)));
-            holder.updateAfterShuffle();
-          });
-        }
-      }
-
-      if(this.p('debug')) {
-        let msg = ''
-        msg += '\n\n\nOPERATION: \n' + JSON.stringify(a, null, '  ');
-        if(problems.length)
-          msg += '\n\nPROBLEMS: \n' + problems.join('\n');
-        msg += '\n\n\nVARIABLES: \n' + JSON.stringify(variables, null, '  ');
-        msg += '\n\nCOLLECTIONS: \n';
-        for(const name in collections) {
-          msg += '  ' + name + ': ' + collections[name].map(w=>`${w.p('id')} (${w.p('type')})`).join(', ') + '\n';
-        }
-        $('#debugButtonOutput').textContent += msg
-        console.log(msg);
-      } else if(problems.length) {
-        console.log(problems);
-      }
-    }
-
-    if(this.p('debug'))
-      showOverlay('debugButtonOverlay');
-
-    batchEnd();
-  }
-
   evaluateInputOverlay(o, resolve, reject, go) {
```

Comment:
Shouldn't the corresponding input overlay functionality also be moved to widget.js?

Reply:
I added this to the room to test INPUT functionality using a `button`. It doesn't seem to work.

![image](https://user-images.githubusercontent.com/10050028/107945859-3b740400-6f5e-11eb-95d5-a83c5ef4553f.png)

Once that works, could change the widget type from button to something else.